e2e/rdctl.e2e.spec.ts

```diff
@@ -255,71 +255,201 @@ test.describe('Command server', () => {
+
+          expect(settings).toMatchObject({
+            version:                expect.anything(),
+            kubernetes:             expect.anything(),
+            portForwarding:         expect.anything(),
+            images:                 expect.anything(),
+            telemetry:              expect.anything(),
+            updater:                expect.anything(),
+            debug:                  expect.anything(),
+            pathManagementStrategy: expect.anything()
+          });
+        });
+        test("it complains when some parameters aren't specified", async() => {
+          for (let idx = 0; idx < parameters.length; idx += 1) {
+            const partialParameters = parameters.slice(0, idx).concat(parameters.slice(idx + 1));
+            const { stdout, stderr, error } = await rdctl(partialParameters.concat(['list-settings']));
+
+            expect({
+              stdout, stderr, error
+            }).toEqual({
+              error:  expect.any(Error),
+              stderr: expect.stringContaining(`Error: open ${ configFilePath }: no such file or directory`),
+              stdout: ''
+            });
+            expect(stderr).not.toContain('Usage:');
+          }
+        });
+        test.describe('when a non-existent config file is specified', () => {
+          test('it fails even when all parameters are specified', async() => {
+            const tmpDir = await fs.promises.mkdtemp(path.join(os.tmpdir(), 'rd-fake-docker'));
+
+            try {
+              const configFile = path.join(tmpDir, 'config.json');
+              // Do not actually create configFile
+              const { stdout, stderr, error } = await rdctl(parameters.concat(['list-settings', '--config-path', configFile]));
+
+              expect({
+                stdout, stderr, error
+              }).toEqual({
+                error:  expect.any(Error),
+                stderr: expect.stringContaining(`Error: open ${ configFile }: no such file or directory`),
+                stdout: ''
+              });
+              expect(stderr).not.toContain('Usage:');
+            } finally {
+              await fs.promises.rm(tmpDir, { recursive: true });
+            }
+          });
+        });
+      });
+    });
     test('should show settings and nil-update settings', async() => {
       const { stdout, stderr, error } = await rdctl(['list-settings']);
 
-      expect(error).toBeUndefined();
-      expect(stderr).toEqual('');
-      expect(stdout).toMatch(/"kubernetes":/);
+      expect({
+        stdout, stderr, error
+      }).toEqual({
+        error:  undefined,
+        stderr: '',
+        stdout: expect.stringContaining('"kubernetes":'),
+      });
       const settings = JSON.parse(stdout);
 
-      expect(['version', 'kubernetes', 'portForwarding', 'images', 'telemetry', 'updater', 'debug', 'pathManagementStrategy']).toMatchObject(Object.keys(settings));
+      expect(settings).toMatchObject({
```

Comment:
You said you wanted to not have extras, right? Should that be `.toEqual` instead of `.toMatchObject`, then?

(And would you want to do `Object.fromEntries(Object.keys(settings.defaultSettings).map(k => [k, expect.anything()]))` to future-proof?)

Reply:
My intention was to do a sanity-check on `rdctl list-settings`.